### PR TITLE
Suppress implicit features from optional dependencies

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: crates/server/src/server/server_future.rs [hunk 1]
```diff
@@ -302,10 +303,12 @@ impl<T: RequestHandler> ServerFuture<T> {
         let tls_acceptor = Box::pin(tls_server::new_acceptor(cert, chain, key)?);
 
         // for each incoming request...
-        let shutdown = self.shutdown_watch.clone();
+        let shutdown = self.shutdown_token.clone();
```
Comment (bluejekyll) [inline crates/server/src/server/server_future.rs:306]: were these changes intended to come into this PR?

Comment (djc) [reply crates/server/src/server/server_future.rs:306]: See the originating commit: https://github.com/hickory-dns/hickory-dns/pull/2337/commits/f9890b209e9b4de53807eb1cbf80a9841dd0fe04. Apparently we don't have coverage of the `dns-over-openssl` feature in the server crate, so I fixed compilation failures as a side effect of making sure I didn't break anything.

Comment (bluejekyll) [reply crates/server/src/server/server_future.rs:306]: Got it.

**Review by bluejekyll — APPROVED** [review]
